frappe/model/base_document.py

```diff
@@ -18,7 +18,7 @@
 	'bigint': 2 ** 63
 }
 
-DOCTYPES_FOR_DOCTYPE = ('DocType', 'DocField', 'DocPerm', 'DocType Action', 'DocType Link')
+DOCTYPES_FOR_DOCTYPE = {'DocType', 'DocField', 'DocPerm', 'DocType Action', 'DocType Link'}
```

Comment:
I know checking memberships are faster in sets...but for such a tiny iterable, it's such a negligible difference (if there exists any) that it'd warrant a change.

<img width="721" alt="Screenshot 2021-12-27 at 12 48 25 PM" src="https://user-images.githubusercontent.com/36654812/147445762-f3fd0bcc-e080-4d96-95a2-c20a3dc06084.png">

Reply:
I don't think your test is correct: container (set, list or tuple) is being defined every time, but `DOCTYPES_FOR_DOCTYPE` is constant that is defined only once.